packages/state-transition/src/block/processConsolidationRequest.ts

```diff
@@ -30,7 +32,7 @@ export function processConsolidationRequest(
   }
 
   if (isValidSwitchToCompoundRequest(state, consolidationRequest)) {
-    switchToCompoundingValidator(state, sourceIndex);
+    switchToCompoundingValidator(state, sourceIndex, metrics ?? null);
```

Comment:
why do we need `?? null`

```suggestion
    switchToCompoundingValidator(state, sourceIndex, metrics);
```

this should work